graylog2-server/src/main/java/org/graylog/events/processor/EventDefinitionHandler.java

```diff
@@ -287,9 +288,29 @@ private void deleteJobDefinition(JobDefinitionDto jobDefinition, EventDefinition
 
     private void deleteJobDefinitionAndTrigger(JobDefinitionDto jobDefinition, EventDefinitionDto eventDefinition) {
         deleteJobTrigger(jobDefinition, eventDefinition);
+        deleteNotificationJobTriggers(eventDefinition);
         deleteJobDefinition(jobDefinition, eventDefinition);
     }
 
+    /**
+     * Deletes notification triggers for this event.
+     * This mostly servers as a safety net, when a misconfiguration has lead to a high number of outstanding
+     * notifications.
+     * Processing these notifications would unnecessarily block the job scheduler and in some cases consume a lot of
+     * resources.
+     */
+    private void deleteNotificationJobTriggers(EventDefinitionDto eventDefinition) {
+        final DBQuery.Query query = DBQuery.and(
```

Comment:
I think we should only delete runnable triggers here.
Otherwise a currently executed job might get confused.
Also I wonder how the scheduler likes it, if the jobdefinition goes away :thinking: 
But that would be an existing bug.

 sth like this should do it:
`                        DBQuery.is(FIELD_STATUS, JobTriggerStatus.RUNNABLE),`

Reply:
Maybe @bernd can answer that quicker, since he has a deeper knowledge about the job scheduler.
Otherwise I can also check the code tomorrow or next week

Reply:
> Maybe @bernd can answer that quicker, since he has a deeper knowledge about the job scheduler. Otherwise I can also check the code tomorrow or next week

@thll and I talked about this two weeks ago. I also first said that we should only remove `RUNNABLE` triggers. But as @thll described, it might leave triggers around which are in a retry loop. We also don't skip `RUNNING` triggers when deleting or disabling an event definition.

So we thought it's okay to do it that way.